File: SDKs/digitalocean.com/2.0/typescript/src/sdk/models/operations/getdockercredentials.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetDockerCredentialsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=expiry_seconds" })
  expirySeconds?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=read_write" })
  readWrite?: boolean;
}


export class GetDockerCredentials200ApplicationJsonAuthsRegistryDigitaloceanCom extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=auth" })
  auth?: string;
}


export class GetDockerCredentials200ApplicationJsonAuths extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=registry.digitalocean.com" })
  registryDigitaloceanCom?: GetDockerCredentials200ApplicationJsonAuthsRegistryDigitaloceanCom;
}


export class GetDockerCredentials200ApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=auths" })
  auths?: GetDockerCredentials200ApplicationJsonAuths;
}


export class GetDockerCredentials401ApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=message" })
  message: string;

  @SpeakeasyMetadata({ data: "json, name=request_id" })
  requestId?: string;
}


export class GetDockerCredentialsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetDockerCredentialsQueryParams;
}


export class GetDockerCredentialsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  headers: Record<string, string[]>;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getDockerCredentials200ApplicationJSONObject?: GetDockerCredentials200ApplicationJson;

  @SpeakeasyMetadata()
  getDockerCredentials401ApplicationJSONObject?: GetDockerCredentials401ApplicationJson;

  @SpeakeasyMetadata()
  onev211ClicksGetResponses401ContentApplication1jsonSchema?: shared.Onev211ClicksGetResponses401ContentApplication1jsonSchema;
}

File: SDKs/digitalocean.com/2.0/python/src/sdk/models/operations/get_docker_credentials.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import onev2_11_clicks_get_responses_401_content_application_1json_schema as shared_onev2_11_clicks_get_responses_401_content_application_1json_schema


@dataclasses.dataclass
class GetDockerCredentialsQueryParams:
    expiry_seconds: Optional[int] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'expiry_seconds', 'style': 'form', 'explode': True }})
    read_write: Optional[bool] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'read_write', 'style': 'form', 'explode': True }})
    

@dataclass_json
@dataclasses.dataclass
class GetDockerCredentials200ApplicationJSONAuthsRegistryDigitaloceanCom:
    auth: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('auth') }})
    

@dataclass_json
@dataclasses.dataclass
class GetDockerCredentials200ApplicationJSONAuths:
    registry_digitalocean_com: Optional[GetDockerCredentials200ApplicationJSONAuthsRegistryDigitaloceanCom] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('registry.digitalocean.com') }})
    

@dataclass_json
@dataclasses.dataclass
class GetDockerCredentials200ApplicationJSON:
    auths: Optional[GetDockerCredentials200ApplicationJSONAuths] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('auths') }})
    

@dataclass_json
@dataclasses.dataclass
class GetDockerCredentials401ApplicationJSON:
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    message: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('message') }})
    request_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('request_id') }})
    

@dataclasses.dataclass
class GetDockerCredentialsRequest:
    query_params: GetDockerCredentialsQueryParams = dataclasses.field()
    

@dataclasses.dataclass
class GetDockerCredentialsResponse:
    content_type: str = dataclasses.field()
    headers: dict[str, list[str]] = dataclasses.field()
    status_code: int = dataclasses.field()
    get_docker_credentials_200_application_json_object: Optional[GetDockerCredentials200ApplicationJSON] = dataclasses.field(default=None)
    get_docker_credentials_401_application_json_object: Optional[GetDockerCredentials401ApplicationJSON] = dataclasses.field(default=None)
    onev2_11_clicks_get_responses_401_content_application_1json_schema: Optional[shared_onev2_11_clicks_get_responses_401_content_application_1json_schema.Onev211ClicksGetResponses401ContentApplication1jsonSchema] = dataclasses.field(default=None)
    
